lib/hawkular/operations/operations_api.rb

```diff
@@ -215,7 +215,10 @@ def check_pre_conditions(hash = {}, params = [], &callback)
       fail 'callback must have the perform method defined. include Hawkular::Operations' unless
           callback.nil? || callback.respond_to?('perform')
       params.each do |property|
-        fail "Hash property #{property} must be specified" if hash[property].nil?
+        if hash[property].nil?
+          error_message = "Hash property #{property} must be specified"
+          callback.nil? ? fail(ArgumentError, error_message) : callback.perform(:failure, error_message)
```

Comment:
Just for clarify, this should fail and throw an exception only in the following cases:

```
client.invoke_generic_operation(noop) do |on|
  on.success do |_data|
    puts 'This should have failed'
  end
end
```
`client.invoke_generic_operation(noop)
`

but shouldn't fail in this case:
```
client.invoke_generic_operation(noop) do |on|
  on.success do |_data|
    puts 'This should have failed'
  end

  on.failure do |error|
    puts error
  end
end

```
Is that correct? If this is the case, I'll do the changes.